src/docker.go

```diff
@@ -72,14 +73,30 @@ func populateFromDocker(i *integration.Integration, args config.ArgumentList) {
 	cgroupInfo, err := dockerClient.Info(context.Background())
 	exitOnErr(err)
 
-	fetcher, err := raw.NewCgroupFetcher(args.HostRoot, cgroupInfo)
-	exitOnErr(err)
+	var fetcher raw.Fetcher
+	if useDockerAPI(args, cgroupInfo.CgroupVersion) {
+		fetcher = dockerapi.NewFetcher(dockerClient)
+	} else { // use cgroups as source of data
+		fetcher, err = raw.NewCgroupFetcher(args.HostRoot, cgroupInfo)
+		exitOnErr(err)
+	}
 
 	sampler, err := nri.NewSampler(fetcher, dockerClient, args)
 	exitOnErr(err)
 	exitOnErr(sampler.SampleAll(context.Background(), i, cgroupInfo))
 }
 
+func useDockerAPI(args config.ArgumentList, version string) bool {
```

Comment:
I would just pass the flag.
```suggestion
func useDockerAPI(useDockerAPI bool, version string) bool {
```

Reply:
If we pass the flag only, let's use other name `userDockerAPI(useDockerAPI bool` is a bit odd, isn't it?

```go
func useDockerAPI(dockerAPIRequested bool, version string) bool {
```

maybe?

Reply:
ok!